### PR TITLE
Support Buildpacks on Cloud Build

### PR DESCRIPTION
**Pull request by dgageot**
Signed-off-by: David Gageot <david@gageot.net>

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [X] Mentions any output changes.
- [X] Adds documentation as needed: user docs, YAML reference, CLI reference.
- [X] Adds integration tests if needed.

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit test added.
- [ ] User facing changes look good.

**Release Notes**

- Support Buildpacks on Cloud Build

### CODE REVIEW

File: pkg/skaffold/build/gcb/buildpacks.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	cloudbuild "google.golang.org/api/cloudbuild/v1"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+)
+
+func (b *Builder) buildpackBuildSpec(artifact *latest.BuildpackArtifact, tag string) (cloudbuild.Build, error) {
+	steps := []*cloudbuild.BuildStep{
+		{
+			Name: "busybox",
+			Args: []string{"sh", "-c", "chown -R 1000:1000 /workspace /layers $$HOME"},
```
Comment (briandealwis) [inline pkg/skaffold/build/gcb/buildpacks.go]: Should this be 
```suggestion
			Args: []string{"sh", "-c", "chown -R $$CNB_USER_ID:$$CNB_GROUP_ID /workspace /layers $$HOME"},
```

Comment (dgageot) [reply pkg/skaffold/build/gcb/buildpacks.go]: Well, `CNB_USER_ID` is hard coded for now and not available as an env variable on GCB.

File: docs/content/en/docs/_index.md [hunk 2]
```diff
@@ -78,8 +78,12 @@ Skaffold supports the following tools:
...
   - on cloud with [Google Cloud Build](https://cloud.google.com/cloud-build/docs/)
 * [Bazel](https://bazel.build/) locally
-* [Cloud Native Buildpacks](https://buildpacks.io/) locally
-* Custom script locally
+* [Cloud Native Buildpacks](https://buildpacks.io/)
+  - locally with Docker
+  - on cloud with [Google Cloud Build](https://cloud.google.com/cloud-build/docs/)
+* Custom script
+  - locally
+  - in-cluster with [Kaniko](https://github.com/GoogleContainerTools/kaniko)
```
[the remark this answers is not in the document]
Comment (dgageot) [reply docs/content/en/docs/_index.md]: yes, it makes no sense!

File: pkg/skaffold/build/gcb/buildpacks.go (hunk 1 above)
Comment (briandealwis) [reply pkg/skaffold/build/gcb/buildpacks.go]: Oh! Maybe this should use the build image rather than busybox.

Comment (briandealwis) [reply pkg/skaffold/build/gcb/buildpacks.go]: (Thinking of distroless images where we should use the non-root user, which [I believe is 65532](https://github.com/GoogleContainerTools/distroless/blob/b54513ef989c81d68cb27d9c7958697e2fedd2c4/base/base.bzl#L8).)

Comment (dgageot) [reply pkg/skaffold/build/gcb/buildpacks.go]: Excellent idea. I changed that [here](https://github.com/GoogleContainerTools/skaffold/pull/3257)

**Pull request merged**